scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,351 @@
+      }
+      mavenDependencyList.add(mavenDependency.build())
+    }
+    return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+  }
+
+  private fun retrieveThirdPartyMavenDependenciesList(
+    rootPath: String
+  ): List<String> {
+    return BazelClient(File(rootPath).absoluteFile, commandExecutor)
+      .retrieveThirdPartyMavenDepsListForBinary("oppia")
+      .map { dep ->
+        if (dep.startsWith(MAVEN_PREFIX)) dep.substring(MAVEN_PREFIX.length, dep.length) else dep
+      }
+  }
+
+  private fun getDependencyListFromMavenInstall(
+    pathToMavenInstall: String,
+    bazelQueryDepsNames: List<String>
+  ): List<MavenListDependency> {
+    val mavenInstallJsonText =
+      File(pathToMavenInstall).inputStream().bufferedReader().use { it.readText() }
+    val moshi = Moshi.Builder().addLast(KotlinJsonAdapterFactory()).build()
+    val adapter = moshi.adapter(MavenListDependencyTree::class.java)
+    val dependencyTree = adapter.fromJson(mavenInstallJsonText)
+    return dependencyTree?.mavenListDependencies?.dependencyList?.filter { dep ->
+      bazelQueryDepsNames.contains(
+        omitVersionAndReplaceColonsHyphensPeriods(dep.coord)
+      )
+    } ?: listOf<MavenListDependency>()
+  }
+
+  private fun getNodeValue(tag: String, element: Element): String {
+    val nodeList = element.getElementsByTagName(tag)
+    val node = nodeList.item(0)
+    if (node != null) {
+      if (node.hasChildNodes()) {
+        val child = node.getFirstChild()
+        while (child != null) {
+          if (child.getNodeType() === Node.TEXT_NODE) {
+            return child.getNodeValue()
+          }
+        }
+      }
+    }
+    return ""
+  }
+
+  private fun replaceHttpWithHttps(
+    url: String
+  ): String {
+    var httpsUrl = url
+    if (!url.startsWith("https")) {
+      httpsUrl = url.replace("http", "https")
+    }
+    return httpsUrl
+  }
+
+  private fun extractLicenseLinksFromPom(
+    pomText: String
+  ): List<License> {
+    val licenseList = mutableListOf<License>()
+    val builderFactory = DocumentBuilderFactory.newInstance()
+    val docBuilder = builderFactory.newDocumentBuilder()
+    val doc: Document = docBuilder.parse(
+      InputSource(ByteArrayInputStream(pomText.toByteArray(Charsets.UTF_8)))
+    )
+
+    val licenses = doc.getElementsByTagName("license")
+    if (licenses.length > 0) {
+      for (i in 0 until licenses.getLength()) {
+        if (licenses.item(0).getNodeType().equals(Node.ELEMENT_NODE)) {
+          val element = licenses.item(i) as Element
+          val licenseName = getNodeValue("name", element)
+          val licenseLink = replaceHttpWithHttps(getNodeValue("url", element))
+          licenseList.add(
+            License.newBuilder().apply {
+              this.licenseName = licenseName
+              this.originalLink = licenseLink
+            }.build()
+          )
+        }
+      }
+    }
+    return licenseList.toList()
```

Comment:
FYI ``toList`` isn't needed; a MutableList is also a List.

Reply:
Done.